benchmarking/runs/multiple_tokenization_goals.py

```diff
@@ -0,0 +1,188 @@
+import math
+
+import typer
+
+from api.models.enums import ScenarioAttribute, Gender, Gpa, Age, Race
+from benchmarking.data.simulated_data.mock_student_provider import (
+    MockStudentProviderSettings,
+    MockStudentProvider,
+)
+from benchmarking.evaluations.goals import DiversityGoal
+from benchmarking.evaluations.graphing.graph_metadata import GraphData
+from benchmarking.evaluations.graphing.line_graph import line_graph
+from benchmarking.evaluations.graphing.line_graph_metadata import LineGraphMetadata
+from benchmarking.evaluations.metrics.average_gini_index import AverageGiniIndex
+from benchmarking.evaluations.metrics.average_gini_index_multi_attribute import (
+    AverageGiniIndexMultiAttribute,
+)
+from benchmarking.evaluations.metrics.maximum_gini_index import MaximumGiniIndex
+from benchmarking.evaluations.metrics.minimum_gini_index import MinimumGiniIndex
+from benchmarking.evaluations.metrics.priority_satisfaction import PrioritySatisfaction
+from benchmarking.evaluations.scenarios.multiple_tokenization_goals import (
+    MultipleTokenizationGoals,
+)
+from benchmarking.simulation.goal_to_priority import goals_to_priorities
+from benchmarking.simulation.simulation import Simulation
+
+
+def concentrate_multiple_max_3(num_trials: int = 10):
+    """
+    Goal: Run a scenario with multiple tokenization constraints to benchmark the algorithm
+    """
+
+    # Defining our changing x-values (in the graph sense)
+    class_sizes = list(range(50, 501, 50))
+    num_trials = 10
+    ratio_of_european_students = 0.45
+    ratio_of_middle_eastern_students = 0.25
+    ratio_of_african_students = 0.15
+    ratio_of_other_students = (
+        1
+        - ratio_of_european_students
+        - ratio_of_middle_eastern_students
+        - ratio_of_african_students
+    )
+    ratio_of_a_students = 0.25
+    ratio_of_b_students = 0.50
+    ratio_of_c_students = 0.25
+    ratio_of_age_18_students = 0.10
+    ratio_of_age_19_students = 0.40
+    ratio_of_age_20_students = 0.30
+    ratio_of_age_21_students = 0.20
+
+    graph_runtime_dict = {}
+    graph_avg_gini_dict = {}
+    graph_priority_dict = {}
+    graph_dicts = [
+        graph_runtime_dict,
+        graph_avg_gini_dict,
+        graph_priority_dict,
+    ]
+
+    for class_size in class_sizes:
+        print("CLASS SIZE /", class_size)
+
+        number_of_teams = math.ceil(class_size / 5)
```

Comment:
The team size here actually matters quite a bit, just me noting that you might want to be a bit more cognizant of this line making the team_size=5